src/Shared/RoslynUtils/TypeHelper.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.CompilerServices
+{
+    internal static class TypeHelper
+    {
+        /// <summary>
+        /// Checks to see if a given type is compiler generated.
+        /// <remarks>
+        /// The compiler doesn't always annotate every time it generates with the
```

Comment:
>The compiler doesn't always annotate every time it generates with the [](http://example.com/codeflow?start=12&length=69)

If you also check the attribute on containing types you shouldn't need to check for the generated name pattern.

Reply:
Check the member itself first, then containing type, then its containing type, etc.